misc/bazel/pkg.bzl

```diff
@@ -1,4 +1,336 @@
+    zip_filename = zip_filename or name
+    zips = zips or {}
+    pkg_filegroup(
+        name = internal("base"),
+        srcs = srcs,
+        visibility = ["//visibility:private"],
+        **kwargs
+    )
+    for kind in ("generic", "arch"):
+        _extract_pkg_filegroup(
+            name = internal(kind),
+            src = internal("base"),
+            kind = kind,
+            visibility = ["//visibility:private"],
+        )
+        pkg_zip(
+            name = internal(kind + "-zip-base"),
+            srcs = [internal(kind)],
+            visibility = ["//visibility:private"],
+        )
+        _zipmerge(
+            name = internal(kind + "-zip"),
+            base = internal(kind + "-zip-base"),
+            zips = zips,
+            zip_name = zip_filename,
+            zip_prefix = name,
+            kind = kind,
+            visibility = visibility,
+        )
+    _imported_zips_manifest(
+        name = internal("zip-manifest"),
+        zips = zips,
+        visibility = ["//visibility:private"],
+    )
+
+    pkg_install(
+        name = internal("script"),
+        srcs = [internal("generic"), internal("arch")],
+        visibility = ["//visibility:private"],
+    )
+    native.filegroup(
+        # used to locate current src directory
+        name = internal("build-file"),
+        srcs = ["BUILD.bazel"],
+        visibility = ["//visibility:private"],
+    )
+    py_binary(
+        name = internal("installer"),
+        srcs = ["//misc/bazel/internal:install.py"],
+        main = "//misc/bazel/internal:install.py",
+        data = [
+            internal("build-file"),
+            internal("script"),
+            internal("zip-manifest"),
+            "//misc/bazel/internal/ripunzip",
+        ],
+        deps = ["@rules_python//python/runfiles"],
+        args = [
+            "--build-file=$(rlocationpath %s)" % internal("build-file"),
+            "--pkg-install-script=$(rlocationpath %s)" % internal("script"),
+            "--destdir",
+            install_dest,
+            "--ripunzip=$(rlocationpath //misc/bazel/internal/ripunzip)",
+            "--zip-manifest=$(rlocationpath %s)" % internal("zip-manifest"),
+        ],
+        visibility = visibility,
+    )
+    native.filegroup(
+        name = name,
+        srcs = [internal("generic-zip"), internal("arch-zip")],
+    )
+
+strip_prefix = _strip_prefix
+
+def _runfiles_group_impl(ctx):
+    files = []
+    for src in ctx.attr.srcs:
+        rf = src[DefaultInfo].default_runfiles
+        if rf != None:
+            files.append(rf.files)
+    return [
+        DefaultInfo(
+            files = depset(transitive = files),
+        ),
+    ]
+
+_runfiles_group = rule(
+    implementation = _runfiles_group_impl,
+    attrs = {
+        "srcs": attr.label_list(),
+    },
+)
+
+def codeql_pkg_runfiles(*, name, exes, **kwargs):
+    """
+    Create a `codeql_pkg_files` with all runfiles from files in `exes`, flattened together.
```

Comment:
I do not understand really what that means, or why we need it tbh - can you explain this again?

Reply:
if you include a target in a `pkg_files` rule, the runfiles are not included. In our case, that's dynamic libraries requried by the swift library. You can use [`include_runfiles`](https://bazelbuild.github.io/rules_pkg/0.10.0/reference.html#pkg_files-include_runfiles), which however either did not exist yet or I just did not see when I did `swift`'s packaging. Right now using that would not be compatible because the runfiles (mainly dynamic libraries) would go in different paths, while we coded the extractor pack to expect the runfiles all flattened alongside the binary (we have a shell wrapper setting the appropriate `LD_LIBRARY_PATH` value).

At some point (in the context of another PR) I would probably experiment with `include_runfiles`, whether we can get rid of this.